README.md

```diff
@@ -145,7 +165,7 @@ code, include a script or makefile that will do the job.
 
 ## A2.5
 
-«replace this with your answer»
+python CheckSentence.py
```

Comment:
I appreciate you adding both positive and negative tests that actually check the result: thank you.

However, your parser is basically hard coded: I was looking for something table driven (hence "Translate this table into a programming language of your choice").

~~~ py
from functools import reduce
ERROR  = "ERROR"
OK     = "OK"

states = {
  0:  { "the":    1 },
  1:  { "lazy":   2, "smelly":  2, "dog": 3, "cat": 3 },
  2:  { "lazy":   2, "smelly":  2, "dog": 3, "cat": 3 },
  3:  { "ate":    4, "ran":     4 },
  4:  { "slowly": 5, "noisily": 5, "EOI": OK },
  5:  { "EOI":    OK }
}

def getNextState(currentState, nextWord):
    return states.get(currentState,{}).get(nextWord, ERROR)

def stateMachine(sentence):
    return reduce(getNextState, sentence, 0) == OK
~~~
Because you have nice tests, I'll deduct less than I normally do for the implementation.

# -3

Reply:
I really appreciate you providing me with this code so I can learn how to make my code more concise instead of using a million if statements. Very elegantly done!